### PR TITLE
[FLINK-15938][table-planner-blink] Enable state cleanup in background…

### PR DESCRIPTION
**Pull request by libenchao**
… for Streaming[SemiAnti]JoinOperator in blink

## What is the purpose of the change

This pr enables state cleanup in background for Streaming[SemiAnti]JoinOperator in blink. 
Default state cleanup in background has been enabled after 1.10, so this only affects 1.9.X.

## Brief change log
 
Add `cleanupInBackground` to `StateTtlConfig` in `JoinRecordStateViews`.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java [hunk 1]
```diff
@@ -79,6 +79,7 @@ static StateTtlConfig createTtlConfig(long retentionTime, boolean stateCleaningE
 				.newBuilder(Time.milliseconds(retentionTime))
 				.setUpdateType(StateTtlConfig.UpdateType.OnCreateAndWrite)
 				.setStateVisibility(StateTtlConfig.StateVisibility.ReturnExpiredIfNotCleanedUp)
+				.cleanupInBackground()
```
Comment (wuchong) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java]: `cleanupInBackground()` is enabled by default and this method is deprecated. Do we add this in master?
From my understanding, we only need to update 1.9 branch.

Comment (libenchao) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java]: Yes. I've explained this in the pr's description, and +1 to only fix for 1.9 branch.
Do you mean I need to open the pr against 1.9 branch?

Comment (wuchong) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java]: Yes. I think we only need a pull request for 1.9 branch.

Comment (libenchao) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java]: Thanks for your review. I've changed this pr for release-1.9 branch.

**Pull request merged**